website/source/docs/providers/docker/r/registry.html.mdown

```diff
@@ -0,0 +1,55 @@
+---
+layout: "docker"
+page_title: "Docker: docker_registry"
+sidebar_current: "docs-docker-resource-registry"
+description: |-
+  Logs to a Docker registry, downloads and exports a Docker image.
+---
+
+# docker\_registry
+
+Used to declare a private docker registry.
+This can be used alongside [docker\_image](/docs/providers/docker/r/image.html)
+to programmatically use the credentials without having to repeat them.
+
+## Example Usage
+
+```
+# Log to a custom docker registry
+resource "docker_registry" "ubuntu_private" {
+    settings_file = "~/.docker/config.json"
+}
+
+# Find the latest Ubuntu precise image.
+resource "docker_image" "ubuntu" {
+    name = "ubuntu:precise"
+    registry = "${docker_registry.ubuntu_private.configurations}"
+}
+
+# Access it somewhere else with ${docker_image.ubuntu.latest}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `settings_file` - (Optional, string) Path to the docker configuration file.
+* `auths` - (Optional, block) See [Auths](#auths) below for details.
+
+<a id="auths"></a>
+### Auths
+
+`auths` is a block within the configuration that can be repeated to specify
```

Comment:
If each of these blocks represents "one auth" then the usual style is to name it in the singular so that each block appears with a singular heading in the config:

```hcl
auth {
    server_address = "..."
    username = "..."
    password = "..."
}
auth {
    server_address = "..."
    username = "..."
    password = "..."
}
```

I'd agree that it feels a bit weird in the implementation to have a list/set field with a singular name, but it makes sense given how this gets written in the configuration.